performance_test/include/performance_test/performance_node_base_impl.hpp

```diff
@@ -324,6 +430,85 @@ void PerformanceNodeBase::topic_callback(
   this->handle_sub_received_msg(topic_name, work_duration, msg->header);
 }
 
+template<typename Action>
+void PerformanceNodeBase::send_action_goal_request(const std::string & name)
+{
+  if (m_action_client_lock) {
+    return;
+  }
+  m_action_client_lock = true;
+
+  // Get client and tracking count from map
+  auto & client_tuple = m_action_clients.at(name);
+  auto client = std::static_pointer_cast<rclcpp_action::Client<Action>>(std::get<0>(client_tuple));
+
+  // Wait for action server to become available
+  if (!client->wait_for_action_server(std::chrono::seconds(1))) {
```

Comment:
this will block the whole node (e.g. maybe there are pubs-subs also that should be running).
I wonder if we should wait 0 here and just skip the request if not discovered.
then the test validation should check how many goals were sent.

maybe the same comment applies also to services